qiskit_ibm_runtime/accounts/account.py

```diff
@@ -0,0 +1,102 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Account related classes and functions."""
+
+
+from typing import Optional
+from urllib.parse import urlparse
+
+from typing_extensions import Literal
+
+AccountType = Optional[Literal["cloud", "legacy"]]
+
+LEGACY_API_URL = "https://auth.quantum-computing.ibm.com/api"
+CLOUD_API_URL = "https://cloud.ibm.com"
+
+
+def _assert_valid_auth(auth: AccountType) -> None:
+    """Assert that the auth parameter is valid."""
+    if not (auth in ["cloud", "legacy"]):
+        raise ValueError(
+            f"Inappropriate `auth` value. Expected one of ['cloud', 'legacy'], got '{auth}'."
+        )
+
+
+def _assert_valid_token(token: str) -> None:
+    """Assert that the token is valid."""
+    if not (isinstance(token, str) and len(token) > 0):
+        raise ValueError(
+            f"Inappropriate `token` value. Expected a non-empty string, got '{token}'."
+        )
+
+
+def _assert_valid_url(url: str) -> None:
+    """Assert that the URL is valid."""
+    try:
+        urlparse(url)
+    except:
+        raise ValueError(f"Inappropriate `url` value. Failed to parse '{url}' as URL.")
+
+
+def _assert_valid_instance(auth: AccountType, instance: str) -> None:
+    """Assert that the instance name is valid for the given account type."""
+    if auth == "cloud":
+        if not (isinstance(instance, str) and len(instance) > 0):
+            raise ValueError(
+                f"Inappropriate `instance` value. Expected a non-empty string."
+            )
```

Comment:
No validation for legacy instance?

Reply:
Added TODO via https://github.com/Qiskit/qiskit-ibm-runtime/pull/49/commits/9fd5a600c08e2b90a5222157d27bdf08e1f834a7